src/NServiceBus.Transport.Msmq/MsmqMessageDispatcher.cs

```diff
@@ -77,6 +79,8 @@ void ExecuteTransportOperation(TransportTransaction transaction, UnicastTranspor
 
                         var label = GetLabel(message);
 
+                        await Task.Yield();
```

Comment:
It just means that you'll be returning to the calling method that much sooner vs. executing everything before the yield on the calling thread.